### PR TITLE
[BTS-1553][3.9] Fix wrong limit value use in LimitExecutor

### PR DESCRIPTION
**Pull request by hkernbach**
### Scope & Purpose

3.9 Backport of: https://github.com/arangodb/arangodb/pull/19592
See details there.

### CODE REVIEW

**Review by mchacki — APPROVED** [review]
LGTM

File: tests/Aql/LimitExecutorTest.cpp [hunk 1]
```diff
@@ -123,99 +91,57 @@ auto const testingInputLengths = ::testing::Values(
...
+      calls.emplace_back(off, false, hard, AqlCall::LimitType::HARD);
+    }
+  }
+
+  /*
+   * Build fullcount<true> calls
+   */
+  for (auto const& off : myOffsets) {
+    for (auto const& hard : myHardLimits) {
+      // Note that fullCount does only make sense with a hard limit, and
```
Comment (mpoeter) [inline tests/Aql/LimitExecutorTest.cpp:134]: Comment is cut off mid-sentence

**Review by mpoeter — CHANGES_REQUESTED** [review]
Changelog entry is missing

File: tests/Aql/LimitExecutorTest.cpp (hunk 1 above)
Comment (hkernbach) [reply tests/Aql/LimitExecutorTest.cpp]: Thanks resolved.

**Review by mpoeter — APPROVED** [review]

**Pull request merged**